modules/ting_fulltext/ting_fulltext.module

```diff
@@ -67,15 +75,17 @@ function ting_fulltext_object_load($object_id) {
 function ting_fulltext_preprocess_ting_relation(&$variables) {
   /** @var \TingRelation $relation */
   $relation = $variables['relation'];
-
   if (!$variables['online'] && (NULL !== $relation->getObject() && NULL != $relation->getObject()->getTingObject())) {
     /** @var \OpenSearch\OpenSearchTingObject $ting_object */
     $ting_object = $relation->getObject()->getTingObject();
 
     // This will only work for the opensearch search provider, so check what
     // type of object we have.
     if ($ting_object instanceof OpenSearchTingObject && in_array('docbook', $relation->getObject()->getTingObject()->getFormatsAvailable())) {
-      $variables['fulltext_link'] = l(t('See the full-text version'), 'ting/object/' . $relation->uri . '/fulltext');
+      // For performance reasons we load the review in full text as ajax.
+      drupal_add_js(drupal_get_path('module', 'ting_fulltext') . '/js/ting_fulltext.js');
+      drupal_add_js(array('tingFulltext' => array('url' => '/ting/fulltext/ajax/' . $relation->uri)), 'setting');
+      $variables['abstract'] = '<div id="ting-lektor-fulltext">' . $variables['abstract'] . '</div>';
```

Comment:
If the original path `'ting/object/' . $relation->uri . '/fulltext` is obsolete then consider removing it from the codebase altogether.

Reply:
Whether to remove the original path is a design decision. I dont know if there are use cases for it.